src/Raven.Server/Documents/TcpHandlers/SubscriptionConnection.cs

```diff
@@ -1139,20 +1139,23 @@ private async Task<bool> WaitForChangedDocs(Task pendingReply)
 
         private Task SendNoopAck()
         {
-            if (ClusterCommandsVersionManager.CurrentClusterMinimalVersion >= 53_000)
+            if (ClusterCommandsVersionManager.CurrentClusterMinimalVersion < 53_000)
             {
-                return TcpConnection.DocumentDatabase.SubscriptionStorage.AcknowledgeBatchProcessed(
+                return TcpConnection.DocumentDatabase.SubscriptionStorage.LegacyAcknowledgeBatchProcessed(
                     SubscriptionId,
                     Options.SubscriptionName,
                     nameof(Client.Constants.Documents.SubscriptionChangeVectorSpecialStates.DoNotChange),
-                    NonExistentBatch, docsToResend: null);
+                    nameof(Client.Constants.Documents.SubscriptionChangeVectorSpecialStates.DoNotChange));
             }
 
-            return TcpConnection.DocumentDatabase.SubscriptionStorage.LegacyAcknowledgeBatchProcessed(
+            if (Strategy == SubscriptionOpeningStrategy.Concurrent)
+                return SubscriptionConnectionsState.SendNoopAckForConcurrentSubscription();
+
+            return TcpConnection.DocumentDatabase.SubscriptionStorage.AcknowledgeBatchProcessed(
                 SubscriptionId,
                 Options.SubscriptionName,
                 nameof(Client.Constants.Documents.SubscriptionChangeVectorSpecialStates.DoNotChange),
-                nameof(Client.Constants.Documents.SubscriptionChangeVectorSpecialStates.DoNotChange));
+                NonExistentBatch, docsToResend: null);
```

Comment:
Let's not revert this `if` prefer to have minimal changes. (so 6.0 merge will easier :) )

Reply:
moved this method to the subscription state :)